### PR TITLE
Style addition with tests

### LINKED ISSUE #44: Migration: add style column to Users table
As a developer, when I retrieve a user record from the database it should have a style attribute. A new user can be given a new style by taking the style survey, but the column will default to `null` when a style has not yet been set.

### LINKED ISSUE #45: GraphQL Mutation: userStyle
As a developer, I can make a GraphQL mutation call to the backend API that adds a style to the current user's database record.

Mutation format:
```graphql
mutation userStyle($id: Int!, $style: String!) {
  userStyle(id: $id, style: $style) {
    name
  }
}
```

Query variables:
```graphql
{
  "id": 1,
  "style": "cool cool cool"
}
```

### PR DESCRIPTION
**Pull request by ribbansmax**
## Is this a fix or a feature?
- feature

## What is the change/fix?
- add userStyle GraphQL mutation to update a user's record after taking a style preference survey

## Where should the reviewer start?
- issue #44 defines the migration we made, added in `api/src/migrations`
- issue #45 provides syntax for the GraphQL mutation call the frontend will need to make to the backend API
  - relevant GraphQL code is in the `api/src/modules/user` directory
- Jest test is located at `api/src/modules/user/tests` 

## How should this be tested?
- `npm run test` from the command line runs the tests we wrote for the user mutation

## This PR is related to which issues:
- closes #45 
- closes #44

### CODE REVIEW

File: code/api/package.json [hunk 1]
```diff
@@ -12,7 +12,8 @@
     "start": "nodemon src/index.js --exec babel-node",
     "start:prod": "npm run build:prod && npm run start:server",
     "build:prod": "babel src -s -D -d build",
-    "start:server": "node build/index.js"
+    "start:server": "node build/index.js",
+    "test": "jest --watch"
```
Comment (philmccarthy) [inline code/api/package.json:16]: This line describes the jest script that runs with `npm run test`

File: code/api/package.json [hunk 2]
```diff
@@ -23,7 +24,9 @@
     "@babel/node": "7.8.7",
     "@babel/plugin-transform-runtime": "7.9.0",
     "@babel/preset-env": "7.9.5",
-    "nodemon": "2.0.3"
+    "jest": "^26.6.3",
+    "nodemon": "2.0.3",
+    "supertest": "^6.1.3"
```
Comment (philmccarthy) [inline code/api/package.json:29]: `supertest` allows us to make GraphQL requests and analyze the response in testing

File: code/api/src/migrations/20210325163237-add-column-style-on-users.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
+'use strict';
+
+module.exports = {
```
Comment (philmccarthy) [inline code/api/src/migrations/20210325163237-add-column-style-on-users.js:3]: Sequelize is the object relational mapper (ORM) this project uses, which maps database records to instances of classes (`objects` in OOP-speak 😃 )

Comment (philmccarthy) [reply code/api/src/migrations/20210325163237-add-column-style-on-users.js:3]: Similar to how we learned database configuration in Rails, migrations let us `change forward` and maintain a record of how the database has been altered over time. This file contains the migration where we add a `style` column to the `users` table.

File: code/api/src/modules/user/model.js [hunk 4]
```diff
@@ -14,6 +14,9 @@ module.exports = function(sequelize, DataTypes) {
     },
     role: {
       type: DataTypes.TEXT
+    },
+    style: {
```
Comment (philmccarthy) [inline code/api/src/modules/user/model.js:18]: the `model.js` file defines the attributes for instances of User (extension of how the Sequelize ORM maps database records to `objects`)

File: code/api/src/modules/user/mutations.js [hunk 5]
```diff
@@ -27,6 +27,22 @@ export const userSignup = {
   resolve: create
 }
 
+// Add style
```
Comment (philmccarthy) [inline code/api/src/modules/user/mutations.js:30]: This is the GraphQL mutation / API call for adding style to a User. See issue #45 for the GraphQL syntax

File: code/api/src/modules/user/resolvers.js [hunk 6]
```diff
@@ -77,3 +78,18 @@ export async function remove(parentValue, { id }) {
 export async function getGenders() {
   return Object.values(params.user.gender)
 }
+
+// Update style
```
Comment (philmccarthy) [inline code/api/src/modules/user/resolvers.js:82]: the GraphQL resolver handles the database logic for when a `userStyle` mutation request is received

**Review by philmccarthy — APPROVED** [review]
Added a few annotations!

**Review by CJSpohn — APPROVED** [review]
Looks awesome!

**Pull request merged**